Dockerfile

```diff
@@ -14,13 +14,13 @@ FROM --platform=$BUILDPLATFORM tonistiigi/xx:${XX_VERSION} AS xx
 FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-alpine${ALPINE_VERSION} AS build-base-alpine
 ENV GOTOOLCHAIN=local
 COPY --link --from=xx / /
-RUN apk add --no-cache bash clang lld llvm file git
+RUN apk add --no-cache bash clang lld llvm musl-dev file git
 WORKDIR /go/src/github.com/docker/cli
 
 FROM build-base-alpine AS build-alpine
 ARG TARGETPLATFORM
 # gcc is installed for libgcc only
-RUN xx-apk add --no-cache musl-dev gcc
+RUN xx-apk add --no-cache gcc
```

Comment:
I worked around this by not using `cgo` for `gotestsum`, but wondering if we still need this, or something similar, but not sure if `musl-dev` must be installed through `xx` here, or if it's safe to install it using a plain `apk add`

/cc @crazy-max perhaps you have ideas   her